internal/authn/oidc/oidc.go

```diff
@@ -83,26 +83,28 @@ func (oidc *RemoteOidcAuthenticator) Authenticate(requestContext context.Context
 	if !token.Valid {
 		return nil, errInvalidToken
 	}
-
 	claims, ok := token.Claims.(jwt.MapClaims)
 	if !ok {
 		return nil, errInvalidClaims
 	}
 
-	validIssuers := []string{
-		oidc.MainIssuer,
+	if err := jwt.NewValidator(jwt.WithIssuedAt()).Validate(claims); err != nil {
+		return nil, errInvalidToken
 	}
```

Comment:
You can get rid of this block and change line 74 to

```go
jwtParser := jwt.NewParser(
		jwt.WithValidMethods([]string{jwt.SigningMethodRS256.Alg()}),
		jwt.WithIssuedAt(),
	)
```

Also, please add `jwt.WithExpirationRequired(),` and add a test for that.